extensions/Skyhigh173/touch.js

```diff
@@ -0,0 +1,258 @@
+            text: 'is touch available?'
+          },
+          {
+            opcode: 'maxMultiTouch',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'maximum finger count'
+          },
+          '---',
+          {
+            opcode: 'numOfFingers',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'number of fingers',
+          },
+          {
+            opcode: 'numOfFingersID',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'number of fingers ID',
+          },
+          {
+            opcode: 'propOfFinger',
+            blockType: Scratch.BlockType.REPORTER,
+            text: '[PROP] of finger [ID]',
+            arguments: {
+              PROP: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: 'x',
+                menu: 'prop',
+              },
+              ID: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: 1
+              }
+            }
+          },
+          {
+            opcode: 'fingerExists',
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: 'finger [ID] exists?',
+            arguments: {
+              ID: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: 1
+              }
+            }
+          },
+          '---',
+          {
+            opcode: 'touchingFinger',
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: 'touching any finger?',
+            disableMonitor: true,
+            filter: [Scratch.TargetType.SPRITE]
+          },
+          {
+            opcode: 'touchingFingerCount',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'current touching finger count',
+            disableMonitor: true,
+            filter: [Scratch.TargetType.SPRITE]
+          },
+          {
+            opcode: 'touchingFingerID',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'current touching finger [X] ID',
+            disableMonitor: true,
+            filter: [Scratch.TargetType.SPRITE],
+            arguments: {
+              X: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: 1
+              }
+            }
+          },
+        ],
+        menus: {
+          prop: {
+            acceptReporters: true,
+            /**
+             * x: x position
+             * y: y position
+             * dx: change of x position compared to previous frame
+             * dy: change of y position compared to previous frame
+             * sx: avg speed x of finger in a second
+             * sy: avg speed y of finger in a second
+             * duration: time since finger press
+             * force (some devices only): force of finger press
+             */
+            items: ['x', 'y', 'dx', 'dy', 'sx', 'sy', 'duration', 'force'],
+          },
+        },
+      };
+    }
+
+    _setup() {
```

Comment:
put this just under the constructor maybe? will make it a little easier to navigate